### PR TITLE
feat: support secrets in render task definition

### PR DESCRIPTION
**Pull request by yehudacohen**
*Issue #, if available:*
#20 
*Description of changes:*
Add ability to add secrets to task definition. There is already a pull request here #143 which tries to do this, but it isn't backward compatible. This pull request is. It also doesn't add tests which this PR does.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kroehre** [comment]
👍 we'd love to use this feature. What's blocking?

**Comment by cairesvs** [comment]
Hi 👋 Any updates about when this will be merged? I think it's a needed feature 🙏

**Comment by canhassancode** [comment]
Hi there, any update on this?

**Comment by amazreech** [comment]
Hi, thank you so much for your contribution. Apologies on the delay. 
We will be working on reviewing Pull Requests on the repository. In the mean time please ensure that below steps, if not already done, are taken care of in your PR:

1. Verify if PR follows semantic pull request [conventions](https://www.conventionalcommits.org/en/v1.0.0/#specification).

2. Please be sure to run `npm run package` command to update **dist/** folder with latest  dependencies.

3. Resolve merge conflicts on the PR if any.

**Review by nickddowns — APPROVED** [review]

**Comment by banthaherder** [comment]
@yehudacohen great work on this! Looks like it is approved just need a quick fix for merge conflicts. Can't wait to use this 🙌

**Comment by aminzar-access** [comment]
Any updates on this? Cheers!

**Comment by pratikgedia** [comment]
@karanbokil Would you be able to help to get this merged in / reviewed ?

File: index.js [hunk 1]
```diff
@@ -68,9 +68,48 @@ async function run() {
           containerDef.environment.push(variable);
         }
       })
+    if (secrets) {
+        // If environment array is missing, create it
```
Comment (amazreech) [inline index.js:72]: Comment Needs Update, from `environment array` to `secrets array`

File: index.js [hunk 2]
```diff
@@ -68,9 +68,48 @@ async function run() {
...
+          if (trimmedLine.length === 0) { return; }
+          // Split on =
+          const separatorIdx = trimmedLine.indexOf("=");
+          // If there's nowhere to split
+          if (separatorIdx === -1) {
+              throw new Error(
+                `Cannot parse the secret '${trimmedLine}'. Secret pairs must be of the form NAME=valueFrom, 
+                where valueFrom is an arn from parameter store or secrets manager. See AWS documentation for more information: 
+                https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html.`);
+          }
```
Comment (amazreech) [inline index.js:91]: Ideally, this (Lines 80 to Lines 91, I think) could be transferred to a common method as we are using same set of code when fetching `environmentVariables` as well.

File: index.js [hunk 3]
```diff
@@ -68,9 +68,48 @@ async function run() {
...
+          } else {
+            // Else, create
+            containerDef.secrets.push(secret);
+          }
+        })
+      }
     }
 
 
+
```
Comment (amazreech) [inline index.js:112]: nit: unnecessary extra line

File: index.test.js [hunk 4]
```diff
@@ -94,13 +121,14 @@ describe('Render task definition', () => {
         expect(core.setOutput).toHaveBeenNthCalledWith(1, 'task-definition', 'new-task-def-file-name');
     });
 
-    test('renders a task definition at an absolute path, and with initial environment empty', async () => {
+    test('renders a task definition at an absolute path, and with initial environment and secrets empty', async () => {
```
Comment (amazreech) [inline index.test.js:124]: __nit:__ I might have understood this wrong and I see that the naming was always this way but the test name here is not accurate as the secrets and environment provided is not empty?

**Review by amazreech — CHANGES_REQUESTED** [review]

**Comment by pt-vamshi** [comment]
waiting for this feature

**Review by amazreech — APPROVED** [review]

**Comment by kg-aws** [comment]
Hi - We have made a follow up PR #334 to fix the merge issues. We have unforgettably added the original authors to that PR. Closing this PR.

https://github.com/aws-actions/amazon-ecs-render-task-definition/pull/334

**Pull request closed without merging**